chaste_codegen/templates/cvode_model.cpp

```diff
@@ -1,241 +1,11 @@
-        {%- endfor %}
-
-        N_Vector dqs = N_VNew_Serial({{derived_quantities|length}});
-        {%- for quant in derived_quantities %}
-        NV_Ith_S(dqs, {{loop.index0}}) = {{quant.var}};
-        {%- endfor %}
-        return dqs;
-    }{% endif %}
-
-    {%- if jacobian_equations|length > 0 %}
-
-    void {{class_name}}::EvaluateAnalyticJacobian(double {{free_variable.var_name}}, N_Vector rY, N_Vector rDY, CHASTE_CVODE_DENSE_MATRIX rJacobian, N_Vector rTmp1, N_Vector rTmp2, N_Vector rTmp3)
-    {
-        {% for state_var in state_vars %}
-        {%- if state_var.in_jacobian %}double {{ state_var.var }} = {% if loop.index0 == membrane_voltage_index %}(mSetVoltageDerivativeToZero ? this->mFixedVoltage : NV_Ith_S(rY, {{loop.index0}}));{%- else %}NV_Ith_S(rY, {{loop.index0}});{%- endif %}
-        // Units: {{state_var.units}}; Initial value: {{state_var.initial_value}}
-        {% endif %}{%- endfor %}
-
-        {% for equation in jacobian_equations %}
-        const double {{equation.lhs}} = {{equation.rhs}};
-        {%- endfor %}
-
-        // Matrix entries{% for entry in jacobian_entries %}
-        IJth(rJacobian, {{entry.i}}, {{entry.j}}) = {% if membrane_voltage_index == entry.i %}mSetVoltageDerivativeToZero ? 0.0 : ({{entry.entry}});{%- else %}{{entry.entry}};{%- endif %}
-        {%- endfor %}
-    }
-    {%- endif %}
-
-template<>
-void OdeSystemInformation<{{class_name}}>::Initialise(void)
-{
-    this->mSystemName = "{{free_variable.system_name}}";
-    this->mFreeVariableName = "{{free_variable.name}}";
-    this->mFreeVariableUnits = "{{free_variable.units}}";
-
-    {% for ode_info in ode_system_information %}// NV_Ith_S(rY,{{loop.index0}}):
-    this->mVariableNames.push_back("{{ode_info.name}}");
-    this->mVariableUnits.push_back("{{ode_info.units}}");
-    this->mInitialConditions.push_back({{ode_info.initial_value}});
-
-    {% endfor %}{% for param in modifiable_parameters %}// mParameters[{{loop.index0}}]:
-    this->mParameterNames.push_back("{{param["name"]}}");
-    this->mParameterUnits.push_back("{{param["units"]}}");
-
-    {% endfor %}{% for param in derived_quantities %}// Derived Quantity index [{{loop.index0}}]:
-    this->mParameterNames.push_back("{{param["name"]}}");
-    this->mParameterUnits.push_back("{{param["units"]}}");
-
-    {% endfor %}{% for attr in named_attributes %}
-    this->mAttributes["{{attr["name"]}}"] = {{attr["value"]}};
-    {% endfor %}this->mInitialised = true;
-}
-
-
-// Serialization for Boost >= 1.36
-#include "SerializationExportWrapperForCpp.hpp"
-CHASTE_CLASS_EXPORT({{class_name}})
-{% if dynamically_loadable %}extern "C"
-{
-    AbstractCardiacCellInterface* MakeCardiacCell(
-            boost::shared_ptr<AbstractIvpOdeSolver> pSolver,
-            boost::shared_ptr<AbstractStimulusFunction> pStimulus)
-    {
-        return new {{class_name}}(pSolver, pStimulus);
-    }
-
-}{%- endif %}
-#endif // CHASTE_CVODE
+{% include "Normal/cpp/header_comments" %}
```

Comment:
I'd be inclined to put things that are shared between Normal and Cvode in a `shared` folder, rather than one using the other - makes it easier to follow the structure, and to see what is common.